### PR TITLE
chore: Document "PipelineBucket" misnomer to avoid confusion

### PR DESCRIPTION
**Pull request by rclinard-amzn**
I spent a bit too much time reading code & templates not realizing that PipelineBucket was a sort of a misnomer. This change aims to make sure that others reading the code are able to see that it is used for all things that Copilot uploads to S3.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/template/templates/app/cf.yml [hunk 1]
```diff
@@ -119,7 +119,7 @@ Outputs:
     Export:
       Name: {{$app}}-ArtifactKey
   PipelineBucket:
-    Description: Bucket used for CodePipeline to stage resources in.
+    Description: "A bucket used for any Copilot artifacts that must be stored in S3 (pipelines, env files, etc)"
```
Comment (huanjani) [inline internal/pkg/template/templates/app/cf.yml]: nit:
```suggestion
    Description: "A bucket used for any Copilot artifacts that must be stored in S3 (pipelines, env files, etc)."
```

File: internal/pkg/deploy/cloudformation/stack/app.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ type AppStackConfig struct {
 type AppRegionalResources struct {
 	Region         string            // The region these resources are in.
 	KMSKeyARN      string            // A KMS Key ARN for encrypting Pipeline artifacts.
-	S3Bucket       string            // S3 bucket for Pipeline artifacts.
+	S3Bucket       string            // A bucket used for any Copilot artifacts that must be stored in S3 (pipelines, env files, etc)
```
Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/app.go]: nit:
```suggestion
	S3Bucket       string            // A bucket used for any Copilot artifacts that must be stored in S3 (pipelines, env files, etc).
```

File: internal/pkg/deploy/cloudformation/stack/app.go [hunk 3]
```diff
@@ -54,13 +54,16 @@ const (
...
 	appExecutionRoleParamName     = "ExecutionRoleName"
 	appDNSDelegationRoleParamName = "DNSDelegationRoleName"
-	appOutputKMSKey               = "KMSKeyARN"
-	appOutputS3Bucket             = "PipelineBucket"
-	appOutputECRRepoPrefix        = "ECRRepo"
-	appDNSDelegatedAccountsKey    = "AppDNSDelegatedAccounts"
-	appDomainNameKey              = "AppDomainName"
-	appDomainHostedZoneIDKey      = "AppDomainHostedZoneID"
-	appNameKey                    = "AppName"
+	// Linked to AppRegionalResources.KMSKeyARN
```
Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/app.go]: We could put the comments on the same lines as the const declarations; might be a bit tidier. Up to you!

**Comment by huanjani** [comment]
Thanks so much for doing this and saving future time! Just some tiny tiny nits.

**Comment by rclinard-amzn** [comment]
Should be fixed now, thanks for the feedback! Having comments on the same line indeed looks nicer.

**Review by huanjani — APPROVED** [review]
LGTM!

File: internal/pkg/deploy/cloudformation/stack/app.go [hunk 4]
```diff
@@ -54,9 +54,9 @@ const (
 	appAdminRoleParamName         = "AdminRoleName"
 	appExecutionRoleParamName     = "ExecutionRoleName"
 	appDNSDelegationRoleParamName = "DNSDelegationRoleName"
-	appOutputKMSKey               = "KMSKeyARN"
-	appOutputS3Bucket             = "PipelineBucket"
-	appOutputECRRepoPrefix        = "ECRRepo"
+	appOutputKMSKey               = "KMSKeyARN"      // Linked to AppRegionalResources.KMSKeyARN.
+	appOutputS3Bucket             = "PipelineBucket" // Linked to AppRegionalResources.S3Bucket.
+	appOutputECRRepoPrefix        = "ECRRepo"        // Linked to AppRegionalResources.RepositoryURLs.
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/stack/app.go]: Can we remove these comments? I'm a bit hesitant to add comments for how things are linked together as those comments can become stale quickly, however feel free to change them to _what_ these consts represent

For example,
```
	appOutputKMSKey               = "KMSKeyARN"      // Name of the CloudFormation Output that holds the KMS Key ARN to encrypt artifact buckets.
```

Comment (rclinard-amzn) [reply internal/pkg/deploy/cloudformation/stack/app.go]: Sure, done!

File: internal/pkg/deploy/cloudformation/stack/app.go [hunk 5]
```diff
@@ -54,9 +54,9 @@ const (
 	appAdminRoleParamName         = "AdminRoleName"
 	appExecutionRoleParamName     = "ExecutionRoleName"
 	appDNSDelegationRoleParamName = "DNSDelegationRoleName"
-	appOutputKMSKey               = "KMSKeyARN"
-	appOutputS3Bucket             = "PipelineBucket"
-	appOutputECRRepoPrefix        = "ECRRepo"
+	appOutputKMSKey               = "KMSKeyARN"      // Name of the CloudFormation Output that holds the KMS Key ARN to encrypt artifact buckets.
+	appOutputS3Bucket             = "PipelineBucket" // Name of the CloudFormation Output that holds the Artifact Bucket ARN for any artifacts that must be stored in S3.
+	appOutputECRRepoPrefix        = "ECRRepo"        // Name of the CloudFormation Output that holds the ECR image repository URL for each service.
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/stack/app.go]:
```suggestion
	appOutputS3Bucket             = "PipelineBucket" // Name of the CloudFormation Output that holds the Artifact Bucket name.
	appOutputECRRepoPrefix        = "ECRRepo"        // Prefix of the CloudFormation Output name that holds the ECR image repository ARN for each service.
```

**Review by efekarakus — APPROVED** [review]

**Pull request merged**